doc/services/pm/device.rst

```diff
@@ -20,90 +27,161 @@ in power saving mode. This method allows saving power even when the CPU is
-* The device's supported transitions between power states.
-* The device's necessary operations to handle the transition between power states.
+    digraph {
+        node [shape=box];
+	rankdir=LR;
 
-The following are some examples of operations that the device driver may perform
-in transition between power states:
+        SUSPENDED [label=PM_DEVICE_STATE_SUSPENDED];
+        ACTIVE [label=PM_DEVICE_STATE_ACTIVE];
+        SUSPENDING [label=PM_DEVICE_STATE_SUSPENDING];
+        OFF [label=PM_DEVICE_STATE_OFF];
 
-* Save/Restore device states.
-* Gate/Un-gate clocks.
-* Gate/Un-gate power.
-* Mask/Un-mask interrupts.
 
-.. note::
+        SUSPEND [label=PM_DEVICE_ACTION_SUSPEND];
+        RESUME [label=PM_DEVICE_ACTION_RESUME];
+        TURN_OFF [label=PM_DEVICE_ACTION_TURN_OFF];
+        TURN_ON [label=PM_DEVICE_ACTION_TURN_ON];
+        FORCE_SUSPEND [label=PM_DEVICE_ACTION_FORCE_SUSPEND];
 
-   When using :ref:`pm-system`, device transitions can be run from the idle thread.
-   As functions in this context cannot block, transitions that intend to use blocking
-   API's **must** check whether they can do so with :c:func:`k_can_yield`.
+	SUSPEND -> SUSPENDING -> SUSPENDED;
+	FORCE_SUSPEND -> SUSPENDING;
+	RESUME -> ACTIVE;
+	TURN_OFF -> OFF;
+	TURN_ON -> ACTIVE;
+    }
+
+As mentioned above, device drivers do not directly change between these states.
+This is entirely done by the power management subsystem. Instead, drivers are
+responsible for implementing any hardware-specific tasks needed to handle state
+changes.
 
 Device Model with Power Management Support
 ******************************************
 
-Drivers initialize the devices using macros. See :ref:`device_model_api` for
-details on how these macros are used. Use the DEVICE_DEFINE macro to initialize
-drivers providing power management support via the PM control function.
-One of the macro parameters is the pointer to the PM action callback.
-If the driver doesn't implement any power control operations, it can initialize
-the corresponding pointer with ``NULL``.
+Drivers initialize devices using macros. See :ref:`device_model_api` for
+details on how these macros are used. A driver which implements device power
+management support must provide these macros with arguments that describe its
+power management implementation.
+
+Use :c:macro:`PM_DEVICE_DEFINE` or :c:macro:`PM_DEVICE_DT_DEFINE` to define the
+power management resources required by a driver. These macros allocate the
+driver-specific state which is required by the power management subsystem.
+
+Drivers can use :c:macro:`PM_DEVICE_GET` or :c:macro:`PM_DEVICE_DT_GET`
+to get a pointer to this state. These pointers should be passed to
+``DEVICE_DEFINE`` or ``DEVICE_DT_DEFINE`` to initialize the
+``pm_device`` field in each ``struct device``.
+
+Here is some example code showing how to implement device power management
+support in a device driver.
+
+.. code-block:: c
+
+    #define DUMMY_DEVICE DT_NODELABEL(dummy_device)
```

Comment:
let's use instance based example, this is not the most common way to define a device driver nowadays.